catalog/core/catalog-core-migratable/src/main/java/org/codice/ddf/catalog/migratable/impl/CatalogMigratableImpl.java

```diff
@@ -76,18 +71,28 @@
      * @param description   description of this migratable
      * @param framework     framework used to retrieve the metacards to export
      * @param filterBuilder builder used to create query filters
-     * @param fileWriter    object used to write metacards to disk
      * @param config        export configuration information
      */
     public CatalogMigratableImpl(@NotNull String description, @NotNull CatalogFramework framework,
             @NotNull FilterBuilder filterBuilder, @NotNull MigrationFileWriter fileWriter,
-            @NotNull CatalogMigratableConfig config) {
-        this(description,
-                framework,
-                filterBuilder,
-                fileWriter,
-                config,
-                new MigrationTaskManager(config, fileWriter));
+            @NotNull CatalogMigratableConfig config, @NotNull String version, @NotNull String id,
+            @NotNull String title, @NotNull String organization) {
+
+        super(version, id, title, description, organization);
+
+        this.description = description;
+        this.framework = framework;
+        this.filterBuilder = filterBuilder;
+        this.config = config;
+        this.taskManager = new MigrationTaskManager(config, fileWriter);
+
+        notNull(description, "description cannot be null");
```

Comment:
`notNull` checks should come before we assign to the fields, right after the `super()` call.